include/items.php

```diff
@@ -845,15 +846,26 @@ function item_store($arr,$force_parent = false, $notify = false, $dontcache = fa
 		}
 	} else {
 		// This can happen - for example - if there are locking timeouts.
-		logger("Item wasn't stored - we quit here.");
-		q("COMMIT");
+		q("ROLLBACK");
+
+		// Store the data into a spool file so that we can try again later.
```

Comment:
not a showstopper but I want to give the suggestion that we should start to create separate functions for some of its parts. `function item_store` is becoming bigger and bigger. Maybe we should think about a own storing class